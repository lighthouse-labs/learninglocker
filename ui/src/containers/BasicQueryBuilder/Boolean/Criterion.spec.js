import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { withInsertCSS } from 'ui/utils/hocs';
import { Map } from 'immutable';
import { initialSections } from 'ui/redux/modules/queryBuilder';
import CriterionWithStyles, { Criterion } from './Criterion.js';

const WrappedCriterion = withInsertCSS(CriterionWithStyles);

test('Criterion should render', () => {
  const section = initialSections.getIn(['result', 'children', 'success']);

  const criterion = ReactTestRenderer.create(
    <WrappedCriterion
      section={section} />
  ).toJSON();

  expect(criterion).toMatchSnapshot();
});

test('Criterion with value', () => {
  const section = initialSections.getIn(['result', 'children', 'success']);

  const mockCriteria = new Map({
    'statement.result.success': true
  });

  const criterion = ReactTestRenderer.create(
    <WrappedCriterion
      criterion={mockCriteria}
      section={section} />
  ).toJSON();

  expect(criterion).toMatchSnapshot();
});

test('Criterion value change', () => {
  const onCriterionChange = jest.fn().mockReturnValueOnce(true);

  const testCriterion = new Criterion({
    criterion: new Map({
      $comment: 'A comment',
    }),
    section: initialSections.getIn(['result', 'children', 'success']),
    onCriterionChange,
    onDeleteCriterion: jest.fn()
  });


  testCriterion.handleValueChange({
    target: {
      value: 'true'
    }
  });

  expect(onCriterionChange.mock.calls.length).toEqual(1);
  expect(onCriterionChange.mock.calls[0][0].get('statement.result.success')).toEqual(true);
});
