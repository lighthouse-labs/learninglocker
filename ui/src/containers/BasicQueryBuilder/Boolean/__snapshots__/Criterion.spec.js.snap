// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Criterion should render 1`] = `
<div
  className="criterion"
>
  <div
    className="criterionValue noCriteria"
  >
    <div>
      <label
        className="radioLabel"
        htmlFor="statement.result.success-1"
      >
        True:  
      </label>
      <input
        checked={false}
        id="statement.result.success-1"
        name="statement.result.success"
        onChange={[Function]}
        type="radio"
        value="true"
      />
    </div>
    <div>
      <label
        className="radioLabel"
        htmlFor="statement.result.success-2"
      >
        False:  
      </label>
      <input
        checked={false}
        id="statement.result.success-2"
        name="statement.result.success"
        onChange={[Function]}
        type="radio"
        value="false"
      />
    </div>
  </div>
</div>
`;

exports[`Criterion with value 1`] = `
<div
  className="criterion"
>
  <div
    className="criterionValue noCriteria"
  >
    <div>
      <label
        className="radioLabel"
        htmlFor="statement.result.success-3"
      >
        True:  
      </label>
      <input
        checked={false}
        id="statement.result.success-3"
        name="statement.result.success"
        onChange={[Function]}
        type="radio"
        value="true"
      />
    </div>
    <div>
      <label
        className="radioLabel"
        htmlFor="statement.result.success-4"
      >
        False:  
      </label>
      <input
        checked={false}
        id="statement.result.success-4"
        name="statement.result.success"
        onChange={[Function]}
        type="radio"
        value="false"
      />
    </div>
  </div>
</div>
`;
